ta/pkcs11/src/processing_ec.c

```diff
@@ -420,45 +420,35 @@ enum pkcs11_rc load_tee_ec_key_attrs(TEE_Attribute **tee_attrs,
 
 enum pkcs11_rc pkcs2tee_algo_ecdsa(uint32_t *tee_id,
 				   struct pkcs11_attribute_head *proc_params,
-				   struct pkcs11_object *obj)
+				   struct pkcs11_object *obj __unused)
 {
 	switch (proc_params->id) {
 	case PKCS11_CKM_ECDSA:
+		/*
+		 * FIXME
+		 * ECDSA without hashing. The TEE algo to use is unknown at this
+		 * point.
+		 */
```

Comment:
Here you could use the key size to define `TEE_ALG_ECDSA_SHAx` for the largest hash supported by the key (as done for `PKCS11_CKM_ECDSA` before this patch. Selected TEE algo will be used only to set/check hash/signature size (done in [`step_asym_operation()`](https://elixir.bootlin.com/op-tee/3.20.0/source/ta/pkcs11/src/processing_asymm.c#L679)). The ECDSA part, based on hash/sign. size, is done later in that function ([signing](https://elixir.bootlin.com/op-tee/3.20.0/source/ta/pkcs11/src/processing_asymm.c#L777) and [verifying](https://elixir.bootlin.com/op-tee/3.20.0/source/ta/pkcs11/src/processing_asymm.c#L787)).

Reply:
Actually, we could set `TEE_ALG_ECDSA_SHA512` for `PKCS11_CKM_ECDSA`. It should do the job.

`xtest pkcs11_1019` currently fails with this P-R because of below switch cases. `pkcs11_1019` uses a 384bit ECC key for several ECDSA flavors but our OP-TEE implementation mandates `ECDSA_SHAx` flavor exactly matches the ECC key size. For example, `TEE_ALG_ECDSA_SHA1` mandates a 192 bit key:
https://github.com/OP-TEE/optee_os/blob/fe2fd3ff46c0bada3dc9b2ae8c1025bfd8f57f3f/lib/libutee/tee_api_operations.c#L72-L77
Maybe the code could be modified:
```patch
-		if (maxKeySize != 192)
+ 		if (maxKeySize < 192)
 			return TEE_ERROR_NOT_SUPPORTED;
```

Reply:
hmmm... I think i got confused in ECDSA_SHA1/P192 IDs and meanings.
I think the right way is:
```c
 case TEE_ALG_ECDSA_SHA1: 
 	if (maxKeySize < 192) 
 		return TEE_ERROR_NOT_SUPPORTED; 
 	break; 

 case __OPTEE_ALG_ECDSA_P192: 
 case __OPTEE_ALG_ECDH_P192: 
 	if (maxKeySize != 192) 
 		return TEE_ERROR_NOT_SUPPORTED; 
 	break;

Reply:
The only constraints I see in the spec on key size for `TEE_ALG_ECDSA_SHA*` is 160 <= size <= 521 (bits) so that's what I implement in the latest update (as well as your suggestion for the PKCS#11 TA). Let's see if it passes CI.